src/checkmatelib/client.py

```diff
@@ -17,10 +19,14 @@ class CheckmateClient:
     def __init__(self, host):
         """Initialise a client for contacting the Checkmate service.
 
+        CHECKMATE_API_KEY must be present to configure authentication
+
         :param host: The host including scheme, for the Checkmate service
         """
         self._host = host.rstrip("/")
 
+        self.api_key = os.environ["CHECKMATE_API_KEY"]
```

Comment:
It might be nicer to have the app do this, and just accept this as a normal parameter:

* This means no implicit behavior from this client (what you see is what you get)
* Testing is simpler
* The apps are in control of how they store and provide the value, and can choose to name it something different if they need to

Reply:
I don't think a library reading basic config / credentials it's always a bad idea necessary as it can simplify operations but I initially allowed both approaches (see the snippet here: https://github.com/hypothesis/checkmatelib/pull/6#discussion_r564400661  but settled on just the environment variable as per this discussion https://github.com/hypothesis/checkmatelib/pull/6#discussion_r564397912 

That being said a I see the benefits of only a function parameter / only environment variable & both and happy to change to either of them but we'll have to agree in one approach beforehand :+1:

Reply:
Either way is fine with me: Either have checkmatelib just read it from a required envvar, or have checkmatelib require it as a Python param and make each app read it from a required envvar.

It is indeed common for libraries to read envvars. Lots of the third-party libs we use do this, lots of `ELASTICSEARCH_*`'s, `NEW_RELIC_*`'s, `SENTRY_*`'s, Gunicorn ones, etc. It does have one advantage of avoiding code duplication (three different apps having to have code for reading this envvar) and avoiding the possibility of confusing variation (if the different apps decide to give the envvar different names or to treat it differently).

Reply:
Well it would avoid apps methods of loading from the environment, but... it would therefore avoid the ways apps are setup to load from the environment _and_ the values passed through paste / gunicorn etc. It also avoids mechanisms those products have for verifying their own params and erroring or reporting appropriately.

I know libraries do this, but that doesn't make it a nice feature. New Relic is horrible and mysterious to use, precisely because you can't control it in any normal way.

If apps want to replicate this behavior they can "opt-in" with `Checkmate(api_key=os.environ["CHECKMATE_API_KEY"], ...)`. It seems unnecessarily opinionated to me for us to force it on them.

Reply:
I'll switch it back to accepting the parameter on the `__init__` then :+1: